src/lifecycle.js

```diff
@@ -188,6 +208,10 @@ function triggerAttributesCreated (target, component) {
 }
 
 function addAttributeListeners (target, component) {
+  if (needsAttrCaching) {
```

Comment:
This won't be necessary if you use `data()` in the `triggerAttributeChanged()` function.